### DISCUSSION TITLE
Homeassistant integration

### DISCUSSION THREAD

**Discussion by tichaonax** [emsesp/EMS-ESP32#232] [Q&A]
Interesting thing you have here, curious about your HA integration. Do you do any customization on HA or its all auto-discovery. Wanted to understand your implementation. If you can point to areas of interest in your code so I can understand it better.

I like to use HA but the learning curve appears steep. If I could do everything through config of auto-discovery feature that is what I want to do.

Thanks!

**Comment by proddy** [comment]
It's all in the code, nothing on the HA side and uses https://www.home-assistant.io/docs/mqtt/discovery/. I would start there and look at the sensor and binary_sensor, experimenting manually using MQTTExplorer or something else to see the entries come in. When that's all working you can start publishing the messages from the C++ code. All my logic is in mqtt.cpp but it's quite complex and very customized to my project.

**Reply by tichaonax** [reply]
Thanks will take a look. That's exactly what l am looking for.

**Reply by varac** [reply]
@tichaonax Did you successfully integrate it into home-assistant ? Do you mind sharing your experiences ?

**Reply by varac** [reply]
for the record: https://emsesp.github.io/docs/#/Home-Assistant
